src/user_interface/BoardPanel.java

```diff
@@ -144,5 +170,8 @@ public void paintComponent(Graphics g) {
                 g2.fillRect(i * tileSize, j * tileSize, tileSize, tileSize);
             }
         }
+
+        paintLegalMoves(g, Color.lightGray, legalMoveCoordinates.get("EMPTY"));
+        paintLegalMoves(g, Color.red, legalMoveCoordinates.get("ATTACK"));
```

Comment:
Maybe change "Empty" and "ATTACK" to Enums in the future. No need to do it now if it works, it works.